Gems/Atom/RPI/Code/Source/RPI.Builders/Model/ModelAssetBuilderComponent.cpp

```diff
@@ -1150,6 +1184,88 @@ namespace AZ
             return AZ::Success(finalMeshList);
         }
 
+        bool ModelAssetBuilderComponent::VertexStreamLayoutMatches(const ProductMeshContent& lhs, const ProductMeshContent& rhs) const
+        {
+            [[maybe_unused]] bool mismatchedVertexLayoutsAreErrors = MismatchedVertexLayoutsAreErrors();
+
+            // Check that the stream counts and types match
+            bool layoutMatches =
+                lhs.m_positions.empty() == rhs.m_positions.empty() &&
+                lhs.m_normals.empty() == rhs.m_normals.empty() &&
+                lhs.m_tangents.empty() == rhs.m_tangents.empty() &&
+                lhs.m_bitangents.empty() == rhs.m_bitangents.empty() &&
+                lhs.m_clothData.empty() == rhs.m_clothData.empty() &&
+                lhs.m_skinJointIndices.empty() == rhs.m_skinJointIndices.empty() &&
+                lhs.m_skinWeights.empty() == rhs.m_skinWeights.empty() &&
```

Comment:
Why the layout match need the data here be empty?

Reply:
These are 1D arrays, so it's just comparing whether, e.g. tangents exist for both meshes or neither mesh. They can be merged as long as both have the vertex data or neither has the vertex data for that particular stream. The size doesn't matter because it's okay for one mesh to have more vertices than the other.

UV sets and colors are 2D arrays since we can have more than one of each stream, so we check that the size (count of the vertex streams) matches.